site/book/04-using-functions/01-declarative-function-execution.md

```diff
@@ -110,13 +110,48 @@ The end result is that:
 If any of the functions in the pipeline fails for whatever reason, then the
 entire pipeline is aborted and the local filesystem is left intact.
 
-## Specifying `image`
+## Specifying `function`
+
+### `image`
 
 The `image` field specifies the container image for the function. You can specify
 an image from any container registry. If the registry is omitted, the default
 container registry for functions catalog (`gcr.io/kpt-fn`) is prepended automatically.
 For example, `set-labels:v0.1` is automatically expanded to `gcr.io/kpt-fn/set-labels:v0.1`.
 
+### `exec`
+
+The `exec` field specifies the executable command for the function. You can specify
+an executable with arguments.
+
+Example below uses `sed` executable to replace all occurances of `foo` with `bar`
+in the package resources.
+
+```yaml
+# PKG_DIR/Kptfile (Excerpt)
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: app
+pipeline:
+  mutators:
+    - exec: "sed -e 's/foo/bar/'"
```

Comment:
It would be better to add an example to declare a normal function executable using exec. Ex: set-labels, apply-setters etc.

Reply:
I was trying to keep the example self-sufficient so that users can try out quickly because `sed` is available on most systems. Using `set-labels` etc. requires additional steps of making those binaries available on the system.

When you say `add an example`, did you mean another example package ?

Reply:
Yeah. It would be useful if you can add the procedure to use function like set-labels. We need not add another example package but a depiction of how to put this feature to work for actual functions. I am fine with merging this PR but we should certainly do it as a followup